plugin/src/test/java/io/jenkins/plugins/casc/misc/ExportImportRoundTripAbstractTest.java

```diff
@@ -0,0 +1,286 @@
+//                    fail(String.format("The Jenkins configuration exported has a value different from the config used. \nJenkins: %s:%s.\nValue: '%s'\nConfig file: %s:%s.\nValue: '%s'",
+//                            oneCompared.getSource().file, oneCompared.getSource().line, oneCompared.getValue(),
+//                            value.getSource().file, value.getSource().line, value.getValue()));
+//                }
+//                break;
+//        }
+//    }
+
+    private void configureWithResource(String config) throws ConfiguratorException {
+        ConfigurationAsCode.get().configure(this.getClass().getResource(config).toExternalForm());
+    }
+
+    private String getResourceContent(String resourcePath) throws IOException {
+        return IOUtils.toString(getClass().getResourceAsStream(resourcePath),
+                "UTF-8");
+    }
+
+    private void writeToFile(String text, String path) throws FileNotFoundException {
+        File file = new File(path);
+        try(PrintWriter out = new PrintWriter(file)) {
+            out.print(text);
+        }
+    }
+
+    private void putConfigInHome(String config) throws IOException {
+        File configFile = new File(r.home, ConfigurationAsCode.DEFAULT_JENKINS_YAML_PATH);
+
+        writeToFile(config, configFile.getAbsolutePath());
+        assertTrue(ConfigurationAsCode.DEFAULT_JENKINS_YAML_PATH + " should be created", configFile.exists());
+    }
+
+//    private String getJenkinsConfViaWebUI() throws Exception {
+//        return download("configuration-as-code/export");
+//    }
+
+//    private String download(String url) throws IOException {
+//        JenkinsRule.WebClient client = r.j.createWebClient();
+//        WebRequest request = new WebRequest(client.createCrumbedUrl(url), POST);
+//        WebResponse response = client.loadWebResponse(request);
+//
+//        assertEquals(200, response.getStatusCode());
+//        return response.getContentAsString();
+//    }
+
+    private void assertConfigViaWebUI(String jenkinsConfig) throws Exception {
+        // The UI requires the path to the config file
+        File f = tempFolder.newFile();
+        writeToFile(jenkinsConfig, f.getAbsolutePath());
+
+        // Call the check url
+        JenkinsRule.WebClient client = r.j.createWebClient();
+        WebRequest request = new WebRequest(client.createCrumbedUrl("configuration-as-code/checkNewSource"), POST);
+        NameValuePair param = new NameValuePair("newSource", f.toURI().toURL().toExternalForm());
+        request.setRequestParameters(Collections.singletonList(param));
+        WebResponse response = client.loadWebResponse(request);
+        assertEquals(200, response.getStatusCode());
+        String res = response.getContentAsString();
+        assertThat(res, containsString("The configuration can be applied"));
+    }
+
+    private void applyConfigViaWebUI(String jenkinsConfig) throws Exception {
```

Comment:
Maybe later we could have JCasC methods, e,g, in https://github.com/jenkinsci/java-client-api

Reply:
it would be great